trading/exchanges/exchange_market_status_fixer.py

```diff
@@ -0,0 +1,270 @@
+                    and ExchangeMarketStatusFixer._is_ms_valid(limit_amount[Ecmsc.LIMITS_AMOUNT_MAX.value]) \
+                    and limit_amount[Ecmsc.LIMITS_AMOUNT_MAX.value] > 0:
+                limit_price[Ecmsc.LIMITS_PRICE_MAX.value] = limit_cost[Ecmsc.LIMITS_COST_MAX.value] / \
+                                                            limit_amount[Ecmsc.LIMITS_AMOUNT_MAX.value]
+
+        if Ecmsc.LIMITS_COST_MIN.value in limit_cost and Ecmsc.LIMITS_AMOUNT_MIN.value in limit_amount:
+            if ExchangeMarketStatusFixer._is_ms_valid(limit_cost[Ecmsc.LIMITS_COST_MIN.value]) \
+                    and ExchangeMarketStatusFixer._is_ms_valid(limit_amount[Ecmsc.LIMITS_AMOUNT_MIN.value]) \
+                    and limit_amount[Ecmsc.LIMITS_AMOUNT_MIN.value] > 0:
+                limit_price[Ecmsc.LIMITS_PRICE_MIN.value] = limit_cost[Ecmsc.LIMITS_COST_MIN.value] / \
+                                                            limit_amount[Ecmsc.LIMITS_AMOUNT_MIN.value]
+
+    @staticmethod
+    def _calculate_amounts(market_limit):
+        limit_cost, limit_price, limit_amount = ExchangeMarketStatusFixer._get_markets_limit(market_limit)
+
+        if Ecmsc.LIMITS_COST_MAX.value in limit_cost and Ecmsc.LIMITS_PRICE_MAX.value in limit_price:
+            if ExchangeMarketStatusFixer._is_ms_valid(limit_cost[Ecmsc.LIMITS_COST_MAX.value]) \
+                    and ExchangeMarketStatusFixer._is_ms_valid(limit_price[Ecmsc.LIMITS_PRICE_MAX.value]) \
+                    and limit_price[Ecmsc.LIMITS_PRICE_MAX.value] > 0:
+                limit_amount[Ecmsc.LIMITS_AMOUNT_MAX.value] = limit_cost[Ecmsc.LIMITS_COST_MAX.value] / \
+                                                              limit_price[Ecmsc.LIMITS_PRICE_MAX.value]
+
+        if Ecmsc.LIMITS_COST_MIN.value in limit_cost and Ecmsc.LIMITS_PRICE_MIN.value in limit_price:
+            if ExchangeMarketStatusFixer._is_ms_valid(limit_cost[Ecmsc.LIMITS_COST_MIN.value]) \
+                    and ExchangeMarketStatusFixer._is_ms_valid(limit_price[Ecmsc.LIMITS_PRICE_MIN.value]) \
+                    and limit_price[Ecmsc.LIMITS_PRICE_MIN.value] > 0:
+                limit_amount[Ecmsc.LIMITS_AMOUNT_MIN.value] = limit_cost[Ecmsc.LIMITS_COST_MIN.value] / \
+                                                              limit_price[Ecmsc.LIMITS_PRICE_MIN.value]
+
+    def _calculate_amount(self):
+        amount_log_price = math.log(self.price_example, 10)
+
+        if amount_log_price >= 0:
+            amount_min = 10 ** (self.LIMIT_AMOUNT_MIN_SUP_ATTENUATION - amount_log_price)
+        else:
+            amount_min = 10 ** -(amount_log_price + self.LIMIT_AMOUNT_MIN_ATTENUATION)
+
+        if amount_log_price >= 0:
+            amount_max = 10 ** (self.LIMIT_AMOUNT_MAX_SUP_ATTENUATION - amount_log_price)
+        else:
+            amount_max = 10 ** (-amount_log_price + self.LIMIT_AMOUNT_MAX_MINUS_3_ATTENUATION)
```

Comment:
merge those two if amount_log_price >= 0: conditions for clarity ?

Reply:
How ?